aten/src/ATen/cuda/cub.cuh

```diff
@@ -386,38 +386,57 @@ __global__ void final_scan_kernel(const T* d_in, T* d_out, T* agg, int64_t nelem
 
 }
 
+template <typename T, typename aggT, bool nonzero>
+struct TransformFunctor {
+  __device__ aggT operator()(T value) const {
+    if constexpr (!nonzero) {
+      return value;
+    } else {
+      return (value != T(0)) ? 1 : 0;
+    }
+  }
+};
 
-
-template<int BLOCK_THREADS, int ITEMS_PER_THREAD, typename T>
-__global__ void calc_block_sums(const T * d_in, T * agg, int64_t nelem, int iters_per_cta){
+template<int BLOCK_THREADS, int ITEMS_PER_THREAD, bool nonzero, typename T, typename aggT>
```

Comment:
I'm not going to carefully review if the aggT template parameters are passed around correctly